src/main/java/io/javarig/RandomGenerator.java

```diff
@@ -43,6 +41,21 @@ public <T> T generate(Type type, int minSize, int maxSize) {
         });
     }
 
+    public <T> T generate(Type type, Type... genericTypes) {
+        Type parameterizedType = new ParameterizedTypeImpl(genericTypes, (Class<?>) type, null);
```

Comment:
please remove null and create a parametrizedTypeImpl constructor with the two first arguments